packages/wonder-blocks-dropdown/src/components/__tests__/single-select.test.tsx

```diff
@@ -1646,4 +1647,661 @@ describe("SingleSelect", () => {
+                // Assert
+                expect(validate).toHaveBeenCalledExactlyOnceWith("1");
+            });
+
+            it("should call onValidate prop", async () => {
+                // Arrange
+                const errorMessage = "error";
+                const onValidate = jest.fn();
+                const userEvent = doRender(
+                    <ControlledSingleSelect
+                        validate={() => errorMessage}
+                        onValidate={onValidate}
+                    />,
+                );
+                const opener = await screen.findByRole("button");
+                await userEvent.click(opener);
+
+                // Act
+                await userEvent.click(screen.getByText("item 1"));
+
+                // Assert
+                expect(onValidate).toHaveBeenCalledExactlyOnceWith(
+                    errorMessage,
+                );
+            });
+            it("should be in an error state when validation fails", async () => {
+                // Arrange
+                const userEvent = doRender(
+                    <ControlledSingleSelect validate={() => "Error"} />,
+                );
+                const opener = await screen.findByRole("button");
+                await userEvent.click(opener);
+
+                // Act
+                await userEvent.click(screen.getByText("item 1"));
+
+                // Assert
+                expect(opener).toHaveAttribute("aria-invalid", "true");
+            });
+
+            it("should be in an error state when validation fails with a custom opener", async () => {
+                // Arrange
+                const userEvent = doRender(
+                    <ControlledSingleSelect
+                        validate={() => "Error"}
+                        opener={() => (
+                            <button aria-label="Search" onClick={jest.fn()} />
+                        )}
+                    />,
+                );
+                const opener = await screen.findByLabelText("Search");
+                await userEvent.click(opener);
+
+                // Act
+                await userEvent.click(screen.getByText("item 1"));
+
+                // Assert
+                expect(opener).toHaveAttribute("aria-invalid", "true");
+            });
+        });
+
+        describe("validation on mount", () => {
+            it("should validate on mount if there is a selected value", () => {
+                // Arrange
+                const validate = jest.fn();
+                // Act
+                doRender(
+                    <ControlledSingleSelect
+                        validate={validate}
+                        selectedValue={"1"}
+                    />,
+                );
+                // Assert
+                expect(validate).toHaveBeenCalledExactlyOnceWith("1");
+            });
+            it("should be in an error state on mount if there is an invalid selected value", async () => {
```

Comment:
Good catch! I had a test for checking it's not in an error state when it is mounted with no value. I added a test checking it's not in an error state when it is mounted with a valid value:

https://github.com/Khan/wonder-blocks/blob/a8fd6e3b6e5917764403769a1fbed35b5c846db8/packages/wonder-blocks-dropdown/src/components/__tests__/single-select.test.tsx#L1887-L1905

I also updated the validate function so it would conditionally return an error message!

Reply:
makes sense, thanks!